### PR TITLE
use system package manager for mpi 

### LINKED ISSUE #868: Single file executable has MPI-related issues, can't launch a simulation.
```
florian@florian-M1010:~/Downloads/antaresXpansion-1.3.0-xpress-Ubuntu-20.04-single-file$ ./antares-xpansion-launcher -i ~/Downloads/antaresXpansion-1.3.0-xpress-ubuntu-20.04/examples/SmallTestFiveCandidates
----------------------------------------------------------------
[Pre Antares][INFO 09-07-2024 14:05:58]  Running Antares Xpansion ... 
[Pre Antares][INFO 09-07-2024 14:05:58]  user: florian
[Pre Antares][INFO 09-07-2024 14:05:58]  hostname: florian-M1010
[Pre Antares][INFO 09-07-2024 14:05:58]  Xpansion version: 1.3.0
[Pre Antares][INFO 09-07-2024 14:05:58]  antares simulator version: 8.8
----------------------------------------------------------------
[StudyLocker][INFO 09-07-2024 14:05:58]  /home/florian/Downloads/antaresXpansion-1.3.0-xpress-ubuntu-20.04/examples/SmallTestFiveCandidates is locked
[ConfigLoader][INFO 09-07-2024 14:05:58]  No solver defined in user/expansion/settings.ini. Cbc used
[GeneralDataProcessor][INFO 09-07-2024 14:05:58]  Pre Antares
[AntaresDriver][INFO 09-07-2024 14:05:58]  Launching antares
[AntaresDriver][INFO 09-07-2024 14:06:03]  Antares simulation duration : 0:00:05.118590
[XpansionDriver][INFO 09-07-2024 14:06:03]  Post Antares
[FullRunDriver][INFO 09-07-2024 14:06:03]  Current directory is now: /home/florian/Downloads/antaresXpansion-1.3.0-xpress-ubuntu-20.04/examples/SmallTestFiveCandidates/output/20240709-1405eco-Xpansion/lp
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    opal_init:startup:internal-failure
But I couldn't open the help file:
    /home/runner/work/antares-xpansion/antares-xpansion/vcpkg_installed/x64-linux/share/openmpi/openmpi/help-opal-runtime.txt: No such file or directory.  Sorry!
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    orte_init:startup:internal-failure
But I couldn't open the help file:
    /home/runner/work/antares-xpansion/antares-xpansion/vcpkg_installed/x64-linux/share/openmpi/openmpi/help-orte-runtime: No such file or directory.  Sorry!
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    mpi_init:startup:internal-failure
But I couldn't open the help file:
    /home/runner/work/antares-xpansion/antares-xpansion/vcpkg_installed/x64-linux/share/openmpi/openmpi/help-mpi-runtime.txt: No such file or directory.  Sorry!
--------------------------------------------------------------------------
*** An error occurred in MPI_Init
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
[florian-M1010:810161] Local abort before MPI_INIT completed completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
Traceback (most recent call last):
  File "launch.py", line 51, in <module>
  File "antares_xpansion/driver.py", line 79, in launch
  File "antares_xpansion/full_run_driver.py", line 60, in launch
  File "antares_xpansion/full_run_driver.py", line 73, in run
antares_xpansion.full_run_driver.FullRunExecutionError: ERROR: exited /tmp/_MEIkIUIcC/bin/full_run with status 1
[810117] Failed to execute script 'launch' due to unhandled exception!
```

### PR DESCRIPTION
**Pull request by a-zakir**

### CODE REVIEW

File: vcpkg.json [hunk 1]
```diff
@@ -14,15 +14,18 @@
...
       "name": "boost-mpi",
-      "version>=": "1.81.0"
+      "version>=": "1.81.0",
+      "platform": "windows"
     },
     {
       "name": "boost-program-options",
-      "version>=": "1.81.0"
+      "version>=": "1.81.0",
+      "platform": "windows"
```
Comment (flomnes) [inline vcpkg.json:23]: why ?

File: vcpkg.json [hunk 2]
```diff
@@ -14,15 +14,18 @@
...
     },
     {
       "name": "boost-algorithm",
-      "version>=": "1.81.0"
+      "version>=": "1.81.0",
+      "platform": "windows"
```
Comment (flomnes) [inline vcpkg.json:28]: why ?

File: .github/workflows/sonarcloud.yml [hunk 3]
```diff
@@ -65,7 +66,12 @@ jobs:
           sudo update-alternatives --set cc /usr/bin/gcc
           sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/g++ 30
           sudo update-alternatives --set c++ /usr/bin/g++
-
+      
+      - name: Compile Boost
+        uses: ./.github/workflows/compile-boost
+        with:
+          prefix: "${GITHUB_WORKSPACE}/deps"
```
Comment (flomnes) [inline .github/workflows/sonarcloud.yml:73]: Why ?

File: vcpkg.json (hunk 1 above)
Comment (a-zakir) [reply vcpkg.json:23]: [for linux platform] if we pull boost-mpi from vcpkg, it'll also install mpi --> against the Pr objective

File: .github/workflows/sonarcloud.yml (hunk 3 above)
Comment (a-zakir) [reply .github/workflows/sonarcloud.yml:73]: boost version in our linux platforms are old, that's why im building the same version as windows.

**Review by tbittar — APPROVED** [review]

**Pull request merged**